Potential NullPointerException in DeploymentConfigurationFactory.createInitParameters

### Description of the bug

Spot in the Vaadin Quarkus add-on [nightly build](https://github.com/vaadin/quarkus/actions/runs/10661158290/job/29546246998#step:8:68928).
On application startup, it might happen that `ServletConfig` contains some init parameters with null value, that is however not supported by `Properties` class.

```
Caused by: java.lang.NullPointerException
	at java.base/java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011)
	at java.base/java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:1006)
	at java.base/java.util.Properties.put(Properties.java:1301)
	at java.base/java.util.Properties.setProperty(Properties.java:229)
	at com.vaadin.flow.server.DeploymentConfigurationFactory.createInitParameters(DeploymentConfigurationFactory.java:97)
	at com.vaadin.flow.server.DeploymentConfigurationFactory.createPropertyDeploymentConfiguration(DeploymentConfigurationFactory.java:75)
	at com.vaadin.flow.server.startup.ServletDeployer$StubServletConfig.createDeploymentConfiguration(ServletDeployer.java:139)
	at com.vaadin.flow.server.startup.ServletDeployer.lambda$getServletConfigurations$0(ServletDeployer.java:198)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at com.vaadin.flow.server.startup.ServletDeployer.getServletConfigurations(ServletDeployer.java:197)
	at com.vaadin.flow.server.startup.ServletDeployer.contextInitialized(ServletDeployer.java:148)
	at com.vaadin.flow.server.startup.ServletContextListeners.contextInitialized(ServletContextListeners.java:44)
	at io.undertow.servlet.core.ApplicationListeners.contextInitialized(ApplicationListeners.java:184)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:217)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:186)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$9$1.call(UndertowDeploymentRecorder.java:623)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:251)
```

For example, running the production mode test from `vaadin/quarkus` repository, the offending key are `src-main-java` and `known-classes`; most likely something set by Quarkus or some extensions.

### Expected behavior

Flow should ignore NULL init parameters and the application should start without errors.

### Minimal reproducible example

* Clone [Vaadin Quakus add-on](https://github.com/vaadin/quarkus) repository
* Set Quarkus `quarkus.version` to a recent version, e.g. `3.14.1`
* Build all modules `mvn clean install -DskipTests`
* Enter production mode test modules and run the tests `cd integration-tests/production; mvn verify`
* Execution should fail with the reported NullPointerException

### Versions

- Vaadin / Flow version: 24.4, 24.5 (most likely also older versions)
- Java version: 17
- OS version: Linux

--- a/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/DeploymentConfigurationFactory.java
@@ -21,6 +21,8 @@ import java.util.Enumeration;
 import java.util.Map;
 import java.util.Properties;
 
+import org.slf4j.LoggerFactory;
+
 import com.vaadin.flow.component.UI;
 import com.vaadin.flow.function.DeploymentConfiguration;
 import com.vaadin.flow.server.startup.AbstractConfigurationFactory;
@@ -94,8 +96,13 @@ public class DeploymentConfigurationFactory extends AbstractConfigurationFactory
         for (final Enumeration<String> e = vaadinConfig
                 .getConfigParameterNames(); e.hasMoreElements();) {
             final String name = e.nextElement();
-            initParameters.setProperty(name,
-                    vaadinConfig.getConfigParameter(name));
+            String value = vaadinConfig.getConfigParameter(name);
+            if (value != null) {
+                initParameters.setProperty(name, value);
+            } else {
+                LoggerFactory.getLogger(DeploymentConfigurationFactory.class)
+                        .debug("Ignoring NULL init parameter {}", name);
+            }
         }
 
         readBuildInfo(initParameters, vaadinConfig.getVaadinContext());

--- a/flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java
@@ -2,6 +2,7 @@ package com.vaadin.flow.server;
 
 import jakarta.servlet.ServletConfig;
 import jakarta.servlet.ServletContext;
+
 import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.Field;
@@ -20,12 +21,6 @@ import java.util.Map;
 import java.util.Properties;
 import java.util.Set;
 
-import com.vaadin.flow.component.UI;
-import com.vaadin.flow.di.Lookup;
-import com.vaadin.flow.di.ResourceProvider;
-import com.vaadin.flow.function.DeploymentConfiguration;
-import com.vaadin.flow.server.frontend.FrontendUtils;
-import com.vaadin.flow.server.startup.ApplicationConfiguration;
 import org.apache.commons.io.FileUtils;
 import org.junit.After;
 import org.junit.AfterClass;
@@ -38,13 +33,19 @@ import org.junit.rules.ExpectedException;
 import org.junit.rules.TemporaryFolder;
 import org.mockito.Mockito;
 
-import static java.util.Collections.emptyMap;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.di.Lookup;
+import com.vaadin.flow.di.ResourceProvider;
+import com.vaadin.flow.function.DeploymentConfiguration;
+import com.vaadin.flow.server.frontend.FrontendUtils;
+import com.vaadin.flow.server.startup.ApplicationConfiguration;
 
 import static com.vaadin.flow.server.Constants.VAADIN_SERVLET_RESOURCES;
 import static com.vaadin.flow.server.InitParameters.SERVLET_PARAMETER_PRODUCTION_MODE;
 import static com.vaadin.flow.server.frontend.FrontendUtils.PARAM_TOKEN_FILE;
 import static com.vaadin.flow.server.frontend.FrontendUtils.TOKEN_FILE;
 import static com.vaadin.flow.server.startup.AbstractConfigurationFactory.DEV_FOLDER_MISSING_MESSAGE;
+import static java.util.Collections.emptyMap;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
@@ -184,6 +185,29 @@ public class DeploymentConfigurationFactoryTest {
                 "Unexpected value for heartbeat interval, should be the same as in servlet context parameters",
                 servletConfigHeartbeatIntervalValue,
                 config.getHeartbeatInterval());
+    }
+
+    @Test
+    public void servletConfigParameters_nullValues_ignored() throws Exception {
+        Class<NoSettings> servlet = NoSettings.class;
+
+        Map<String, String> servletConfigParams = new HashMap<>(
+                defaultServletParams);
+        servletConfigParams.put("someKey", null);
+        servletConfigParams.put("someNotNullKey", "NOT_NULL");
+
+        Map<String, String> servletContextParams = new HashMap<>();
+
+        DeploymentConfiguration config = new DeploymentConfigurationFactory()
+                .createDeploymentConfiguration(servlet, createVaadinConfigMock(
+                        servletConfigParams, servletContextParams));
+
+        Assert.assertFalse(
+                "Expecting null parameter to be ignored, but was in configuration",
+                config.getInitParameters().containsKey("someKey"));
+        Assert.assertTrue(
+                "Expecting not null parameter to be in configuration, but was not",
+                config.getInitParameters().containsKey("someNotNullKey"));
     }
 
     @Test

